soh/soh/OTRGlobals.cpp

```diff
@@ -1771,11 +1763,10 @@ extern "C" int CustomMessage_RetrieveIfExists(GlobalContext* globalCtx) {
             } else {
                 messageEntry = Randomizer_GetGanonHintText();
             }
-        } else if (textId >= 0x9000 && textId <= 0x905F) {
-            messageEntry = Randomizer_GetScrubMessage((textId & ((1 << 8) - 1)));
-        } else if (textId >= 0x9100 && textId <= 0x9180) {
-            // TODO: Magic number 100, we don't to overwrite deku scrub messages
-            messageEntry = Randomizer_GetShopMessage((textId & ((1 << 8) - 1)) + 100);
+        } else if (textId >= TEXT_SCRUB_RANDOM && textId <= TEXT_SCRUB_RANDOM + 0xFF) {
+            messageEntry = CustomMessageManager::Instance->RetrieveMessage(Randomizer::merchantMessageTableID, textId);
+        } else if (textId >= TEXT_SHOP_ITEM_RANDOM && textId <= TEXT_SHOP_ITEM_RANDOM + 0xFF) {
+            messageEntry = CustomMessageManager::Instance->RetrieveMessage(Randomizer::merchantMessageTableID, textId);
```

Comment:
`0xFF` is a magic number here now

Reply:
Good point, the top one could be 95, as that's the max price for scrubs, and the bottom could be NUM_SHOP_ITEMS * 2